src/core_codemods/sql_parameterization.py

```diff
@@ -128,52 +153,117 @@ def transform_module_impl(self, tree: cst.Module) -> cst.Module:
+            else:
+                self.changed_nodes[expr] = cst.parse_expression('""')
+
+        prepend = append = None
 
         # remove quote literal from start
-        match start:
+        current_start = self.changed_nodes.get(start) or start
+        prepend_raw_value = None
+
+        t = _extract_prefix_raw_value(self, current_start)
+        prefix, raw_value = t if t else ("", "")
+
+        # gather string after the quote
+        if "r" in prefix:
+            quote_span = list(raw_quote_pattern.finditer(raw_value))[-1]
+        else:
+            quote_span = list(quote_pattern.finditer(raw_value))[-1]
+        new_raw_value = raw_value[: quote_span.start()] + parameter_token
+        prepend_raw_value = raw_value[quote_span.end() :]
+
+        match current_start:
             case cst.SimpleString():
-                current_start = self.changed_nodes.get(start) or start
-                if current_start.raw_value.endswith("\\'"):
-                    new_raw_value = current_start.raw_value[:-2] + parameter_token
-                else:
-                    new_raw_value = current_start.raw_value[:-1] + parameter_token
+                # uses the same quote and prefixes to guarantee it will be correctly interpreted
+                if prepend_raw_value:
+                    prepend = cst.SimpleString(
+                        value=current_start.prefix
+                        + current_start.quote
+                        + prepend_raw_value
+                        + current_start.quote
+                    )
+
                 new_value = (
                     current_start.prefix
                     + current_start.quote
                     + new_raw_value
                     + current_start.quote
                 )
                 self.changed_nodes[start] = current_start.with_changes(value=new_value)
+
             case cst.FormattedStringText():
-                # TODO formatted string case
-                pass
+                if prepend_raw_value:
+                    prepend = cst.SimpleString(
+                        value=("r" if "r" in prefix else "")
+                        + "'"
+                        + prepend_raw_value
+                        + "'"
+                    )
+
+                new_value = new_raw_value
+                self.changed_nodes[start] = current_start.with_changes(value=new_value)
+
+        # remove quote literal from end
+        current_end = self.changed_nodes.get(end) or end
+        append_raw_value = None
+
+        t = _extract_prefix_raw_value(self, current_end)
+        prefix, raw_value = t if t else ("", "")
+        if "r" in prefix:
+            quote_span = list(raw_quote_pattern.finditer(raw_value))[0]
+        else:
+            quote_span = list(quote_pattern.finditer(raw_value))[0]
+
+        new_raw_value = raw_value[quote_span.end() :]
+        append_raw_value = raw_value[: quote_span.start()]
+        match current_end:
```

Comment:
This seems almost exactly the same as the block starting with `match current_start:` above. I might suggest factoring this out into a separate function for readability since this method is already quite long.